cmd/download.go

```diff
@@ -39,14 +39,18 @@ func Download(ctx *cli.Context) {
 	}
 
 	for name, contents := range submission.ProblemFiles {
-		if err := ioutil.WriteFile(fmt.Sprintf("%s/%s", path, name), []byte(contents), 0755); err != nil {
+		if err := ioutil.WriteFile(fmt.Sprintf("%s/%s", path, name), []byte(contents), 0644); err != nil {
 			log.Fatalf("Unable to write file %s: %s", name, err)
 		}
 	}
 
 	for name, contents := range submission.SolutionFiles {
 		filename := strings.TrimPrefix(name, strings.ToLower("/"+submission.TrackID+"/"+submission.Slug+"/"))
-		if err := ioutil.WriteFile(fmt.Sprintf("%s/%s", path, filename), []byte(contents), 0755); err != nil {
+		dir := filepath.Dir(filename)
+		if err := os.MkdirAll(fmt.Sprintf("%s/%s", path, dir), 0755); err != nil {
```

Comment:
I don't know how similar this code path is starting to get with https://github.com/exercism/cli/blob/master/user/item.go#L51 - maybe could use some unification? I'm unsure of where I'd put the code though, so I'd take suggestions on that.